plutus-core/plutus-ir/src/PlutusIR/Transform/EvaluateBuiltins.hs

```diff
@@ -55,18 +55,18 @@ evaluateBuiltins conservative binfo costModel = transformOf termSubterms process
       -> Maybe (Term tyname name uni fun ())
     eval (BuiltinCostedResult _ getX) AppContextEnd =
         case getX of
-            BuiltinSuccess v           -> Just v
+            BuiltinSuccess term           -> Just term
             -- Evaluates successfully, but does logging. If we're being conservative
             -- then we should leave these in, so we don't remove people's logging!
             -- Otherwise `trace "hello" x` is a prime candidate for evaluation!
-            BuiltinSuccessWithLogs _ v -> if conservative then Nothing else Just v
+            BuiltinSuccessWithLogs _ term -> if preserveLogging then Nothing else Just term
```

Comment:
It perhaps should be `conservative` rather than `preserveLogging`, because we probably don't want to optimize `error` that is on the evaluation path for example. But maybe not, I asked something along these lines on Slack.

Reply:
The thing is that at the `Compiler` applies this `evaluateBuiltins` function like this:
```haskell
preserveLogging <- view (ccOpts . coPreserveLogging)
... $ EvaluateBuiltins.evaluateBuiltinsPass tcconfig preserveLogging binfo costModel
```